com.unity.shadergraph/Editor/Util/MessageManager.cs

```diff
@@ -143,48 +166,91 @@ void DebugPrint()
             Debug.Log(output.ToString());
         }
 
-        public static void Log(AbstractMaterialNode node, string path, ShaderMessage message, Object context)
+        public static void Log(AbstractMaterialNode node, string path, ShaderMessage message, Object context, IErrorLog log = null)
```

Comment:
Not sure if this is actually needed, but I maintained backwards compatibility on this function -- it will go back to debug.log if no override is passed to the interface.

Reply:
Definitely better to be safe.  Might be good to make a pass over the logging later to update any stragglers to the new interface.